front/src/components/SiteForm/SearchTemplate.tsx

```diff
@@ -27,18 +35,20 @@ const default_nctid = 'NCT00222898';
 
 function SearchTemplate(props: Props) {
   const [nctId, setNctId] = useState(default_nctid);
-  const { data } = useQuery<StudyPagePrefetchQuery>(PREFETCH_QUERY, {
-    variables: { nctId },
+  const [fragmentState, setFragment] = useState('');
+  const { data: introspection } = useQuery<IntrospectionQuery>(
+    gql(getIntrospectionQuery({ descriptions: false }))
+  );
+  const [fragmentName, fragment] = useFragment('Study', props.template);
+  const { data: study } = useQuery(getStudyQuery(fragmentName, fragment), {
+    variables: { 
+      nctId: nctId
+     },
   });
-  const schema: SchemaType = useMemo(()=> ({
-    kind: 'json',
-    schema: {
-      type: 'object',
-      properties: fromPairs(
-        props.fields.map(f => [camelCase(f), { type: 'string' }])
-      )
-    },
-  }), [props.fields]);
+  if (!introspection) {
+    return <BeatLoader />;
+  }
+  const types = introspection.__schema.types;
   return (
     <Container>
       <StyledFormControl

```

Comment:
Could this SearchTemplate make use of the MailMergeFormControl to avoid having do to the introspection query itself?

Reply:
@rodctech confirming you get this